### DISCUSSION TITLE
WinMergeU - CLI - How to get the output of two file comparision in command line arguments?

### DISCUSSION THREAD

**Discussion by pcakhilnadh** [WinMerge/winmerge#731] [Q&A] [answered]
Hi There,

I am trying to compare two text documents based on the [documentation](https://manual.winmerge.org/en/Command_line.html) using command-line arguments as follows

`WinMergeU file1\t1.txtfile2\t1.txt`

The output opens in a UI screen. I wanted the comparison result to be shown on the commandline or in a output file.
Is it possible by any menas?

**Comment by sdottaka** [comment, marked as answer]
WinMerge can output the comparison result to an HTML file. To output the comparison result to an HTML file, use the -or option as shown below.

~~~
"c:\Program Files\WinMerge\WinMergeU.exe" file1\t1.txt  file2\t1.txt -or c:\tmp\report.html
~~~

If you don't want to see the WinMerge window, specify -noninteractive as follows:

~~~
"c:\Program Files\WinMerge\WinMergeU.exe" -noninteractive file1\t1.txt  file2\t1.txt -or c:\tmp\report.html
~~~

**Reply by pcakhilnadh** [reply]
Wow that is great !  Thank you.
![out](https://user-images.githubusercontent.com/3054812/114977159-de84c380-9ea4-11eb-93ca-ca258e55fcea.png)
In the [documentation ](https://manual.winmerge.org/en/Command_line.html) it is written `-o` .  It should be a typo then right?

**Reply by sdottaka** [reply]
The /or option is still an undocumented option. It is different from the /o option.

**Reply by TrilokSingh457** [reply]
Thank you @sdottaka I have tried the /or option it worked fine. I noticed the output was CSV format so instead of .html as the output file type, I have used .csv file instead, it was even better.

**Reply by sergeevabc** [reply]
I'm not happy with that `-or out.html` solution. I asked Winmerge to compare two 40 MB text files and got 200+ MB html file. Too long, too resource-hungry. And the difference between these files is only a couple of characters, that's not funny.

**Reply by sdottaka** [reply]
You can set DiffContext to display/output only the lines with differences. Please refer to the following.
https://github.com/WinMerge/winmerge/discussions/2117

**Reply by sergeevabc** [reply]
@sdottaka, indeed, adding `/cfg DiffContextV2=0` (what an unmemorable key, now I have to carry the manual around) did the trick, the diff report was reduced from 200 MB to 0,01 MB. But it is so stretched across the screen that it is not particularly convenient to work with. Is it possible to output diff not side by side, but underneath each other, as in the following example done by [riff](https://github.com/walles/riff)?

![2024-0126-0207 riff original patched](https://github.com/WinMerge/winmerge/assets/373718/ebe675ca-f321-47ac-8a8b-98bc7e225215)

**Reply by sdottaka** [reply]
You can output in a riff-like format from the Tools → Generate Patch menu item, but unfortunately you cannot output from the command line.
![image](https://github.com/WinMerge/winmerge/assets/98126/61be5681-e2e7-46f9-8b39-e708bf54773e)

**Reply by sergeevabc** [reply]
@sdottaka, for some unknown reason WinMerge outputs the patch in ANSI instead of UNICODE, and at the same time changes some chars in the filename (e.g. vögel.txt becomes vogel.txt).

``` console
--- C:/birds.txt	Thu Jan 25 23:37:19 2024
+++ C:/vogel.txt	Thu Jan 25 23:38:40 2024
```

**Reply by sdottaka** [reply]
The problem may be resolved by setting the default codepage to 65001 (UTF-8) in the Options window as shown below.

![image](https://github.com/WinMerge/winmerge/assets/98126/085fcad4-41ea-47be-8d69-1ee4f51b675d)

**Comment by Meena1983** [comment]
@sdottaka I am looking for a solution where I can compare two folders and get the individual file comparison reports in another folder. I am unable to find options for this.

"C:\Program files\winmerge\WinMergeU.exe"   "C:\Users\12513\Desktop\source"  "C:\Users\12513\Desktop\modified" /r /e /x /s /u /wl /e /unpacker "PrettifyXML" /maximize /minimize /or "C:\Users\12513\Desktop\Result\compare.htm"  

Can you please let me know if there are any options?

**Reply by sdottaka** [reply]
Answered https://github.com/WinMerge/winmerge/discussions/2034.
Unfortunately /unpacker "PrettifyXML" does not work for folder comparisons.

**Reply by Meena1983** [reply]
Thank you! I will work on comparing files individually.

**Comment by HarshalNCT** [comment]
Are there any pre-requisite for using winmerge on commandline @sdottaka ? because I tried this command "C:\Program Files (x86)\WinMerge\WinMergeU.exe" -noninteractive  D:\Projects\document1.txt D:\Projects\document2.txt -or  D:\Projects\comparison_report.html but it still opens winmerge UI and report doesn't gets created or saved

**Reply by sdottaka** [reply]
In my environment, this command line `"C:\Program Files (x86)\WinMerge\WinMergeU.exe" -noninteractive D:\Projects\document1.txt D:\Projects\document2.txt -or D:\Projects\comparison_report.html` works.

As a prerequisite, your WinMerge version must be at least 2.16.0 or higher.

**Reply by HarshalNCT** [reply]
I have used WinMerge-2.16.42.1-x64-Setup.exe for setup. Is it correct am I missing something ?

**Reply by sdottaka** [reply]
If you installed WinMerge-2.16.42.1-x64-Setup.exe, the installation destination will be "C:\Program Files\WinMerge" instead of "C:\Program Files (x86)\WinMerge", so try changing the path.
